cmake/Tools/Platform/Android/android_deployment.py

```diff
@@ -453,14 +465,21 @@ def install_assets_to_device(self, detected_storage, target_device):
                            device_id=target_device)
             logging.info(f"Device '{target_device}': Target cleaned.")
 
-        # '/.' is necessary to avoid copying folder 'assets' to destination, but its content.
-        assets_layout_src = f'{str(self.local_asset_path)}/.'
-        assets_layout_dst = f'{output_target}'
+        # On certain devices pushing files with adb fails with 'remote secure_mkdirs failed' error.
+        # Creating all dirs first on target to surpass the issue.
+        self.adb_shell(command=f'mkdir {output_package}', device_id=target_device)
+        self.adb_shell(command=f'mkdir {output_target}', device_id=target_device)
+        for asset_path in self.files_in_asset_path:
+            if asset_path.is_dir():
+                relative_path = os.path.relpath(str(asset_path), str(self.local_asset_path)).replace('\\', '/')
+                target_path = f"{output_target}/{relative_path}"
```

Comment:
This should [pathlib.relative_to](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) along with [pathlib.as_posix](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) to convert to forward slashes

Reply:
I used the same code from below that was there before. I'll take a look at your suggestion tomorrow and test it.